src/IBM.Cloud.SDK.Core/Authentication/Iam/IamConfig.cs

```diff
@@ -0,0 +1,100 @@
+﻿/**
+* Copyright 2019 IBM Corp. All Rights Reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*      http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*
+*/
+
+using IBM.Cloud.SDK.Core.Util;
+using System;
+
+namespace IBM.Cloud.SDK.Core.Authentication.Iam
+{
+    /// <summary>
+    /// Options for authenticating via the IAM (Identity Access Management) token service.
+    /// This authentication mechanism allows an IAM access token to be obtained using a
+    /// user-supplied apikey.
+    /// The IAM access token is then specified as a Bearer Token within outgoing REST API requests.
+    /// </summary>
+    public class IamConfig : IAuthenticatorConfig
+    {
+        public string Apikey { get; private set; }
+        public string IamUrl { get; private set; }
+        public string UserManagedAccessToken { get; private set; }
+        public bool? DisableSslVerification { get; private set; }
+        public string IamClientId { get; private set; }
+        public string IamClientSecret { get; private set; }
+        private readonly string defaultUrl = "https://iam.cloud.ibm.com/identity/token";
+
+        public string AuthenticationType
+        {
+            get { return Authenticator.AuthtypeIam; }
+        }
+
+        public void Validate()

```

Comment:
If the user provides an `IamClientId`, they have to provide the `IamClientSecret` as well, and vice-versa. You could validation for that too if you want.

Reply:
good call - added